river/proba/gaussian.py

```diff
@@ -90,3 +95,203 @@ def sample(self):
+    7  0.139494  0.292145  0.366362
+
+    >>> p = MultivariateGaussian()
+    >>> p.n_samples
+    0.0
+    >>> for x in X.to_dict(orient="records"):
+    ...     p = p.update(x)
+    >>> p._var
+            blue     green    red
+     blue    0.076    0.020   -0.010
+    green    0.020    0.113   -0.053
+      red   -0.010   -0.053    0.079
+
+    Weighted samples are currently not implemented. Updates with default w = 1.
+    >>> p = p.update(x, w=2)
+
+    Retrieving current state in nice format is simple
+    >>> p
+    𝒩(μ=(0.385, 0.376, 0.501),
+    σ^2=([0.069 0.019 -0.004]
+     [0.019 0.100 -0.044]
+     [-0.004 -0.044 0.078]))
+    >>> p.n_samples
+    9.0
+
+    To retrieve pdf and cdf
+    >>> p(x)  # doctest: +ELLIPSIS
+    1.70399123552737...
+    >>> p.cdf(x)  # doctest: +ELLIPSIS
+    0.01421620021072799...
+
+    MultivariateGaussian works with Rolling
+    >>> from river import utils
+    >>> p = utils.Rolling(MultivariateGaussian(), window_size=5)
+    >>> for x in X.to_dict(orient="records"):
+    ...     p = p.update(x)
+    >>> p._var
+            blue     green    red
+     blue    0.087   -0.023    0.008
+    green   -0.023    0.014   -0.025
+      red    0.008   -0.025    0.095
+
+    MultivariateGaussian works with TimeRolling
+    >>> from datetime import datetime as dt, timedelta as td
+    >>> X.index = [dt(2023, 3, 28, 0, 0, 0) + td(seconds=x) for x in range(8)]
+    >>> p = utils.TimeRolling(MultivariateGaussian(), period=td(seconds=5))
+    >>> for t, x in X.iterrows():
+    ...     p = p.update(x.to_dict(), t=t)
+    >>> p._var
+            blue     green    red
+     blue    0.087   -0.023    0.008
+    green   -0.023    0.014   -0.025
+      red    0.008   -0.025    0.095
+
+    Weighted samples are currently not implemented. Updates with default w = 1.
+    >>> p = p.update(x.to_dict(), t=t + td(seconds=1), **{"w":2})
+
+    >>> from river.proba import Gaussian
+    >>> p = MultivariateGaussian()
+    >>> p_ = Gaussian()
+    >>> for t, x in X.iterrows():
+    ...     p = p.update(x.to_dict())
+    ...     p_ = p_.update(x['blue'])
+    >>> p.sigma[0][0] == p_.sigma
+    True
+
+    Initiation of class from state is currently not implemented
+    >>> p = MultivariateGaussian()._from_state(
+    ...     0, 0, 0, 0)  # doctest: +IGNORE_EXCEPTION_DETAIL
+    Traceback (most recent call last):
+    ...
+    NotImplementedError: from_state_ not implemented yet.
+
+    """  # noqa: W291
+
+    def __init__(self, seed=None):
+        super().__init__(seed)
+        self._var = covariance.EmpiricalCovariance(ddof=1)
+        self.feature_names_in_ = None
+
+    @classmethod
+    def _from_state(cls, n, m, sig, ddof):
+        raise NotImplementedError("_from_state not implemented yet.")
+
+    @property
+    def n_samples(self):
+        if not self._var.matrix:
+            return 0.0
+        else:
+            return list(self._var.matrix.values())[-1].mean.n
+
+    @property
+    def mu(self):
```

Comment:
This should return a dictionary, not a list

Reply:
Thank you for your valuable suggestions. I made properties "mu" and "mode" return dictionary.